tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/Auth.java

```diff
@@ -155,6 +166,32 @@ public Response info(
         LoginResponse resp = new LoginResponse();
         resp.sessionId = session;
         resp.user = s.user;
-        return Response.ok().entity(resp).build();
+        return Response.ok().entity(resp)
+            .header(SESSION_HEADER, session)
+            .build();
+    }
+
+
+    /**
+     * Extract a CookieSession from a session string
+     * @param session
+     * @return session or null
+     */
+    public static CookieSession getSession(String session) {
+        CookieSession s = CookieSession.retrieveWithoutTouch(session);
+        return s;
+    }
+
+    /**
+     * Assert that a session is valid
+     * @param session
+     * @return null or a Response to return
+     */
```

Comment:
The protocol is a little confusing: null means valid, non-null means invalid

Reply:
it's for the convenience of the calling code, otherwise yes it is confusing